### PR TITLE
getProjectionStatus

### PR DESCRIPTION
**Pull request by George-Payne**
Implement `getProjectionStatus`.

ref: #238

### CODE REVIEW

**Comment by oskardudycz** [comment]
@George-Payne, could you add a snippet for that method like, e.g. in the .NET client? https://github.com/EventStore/EventStore-Client-Dotnet/blob/master/samples/projection-management/Program.cs#L294

File: src/__test__/projections/getProjectionStatus.test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  PREPARED,
+  PREPARING,
+  RUNNING,
+  STARTING,
+  STOPPED,
+  STOPPING,
+  UnknownError,
+} from "@eventstore/db-client";
+
+describe("getProjectionStatistics", () => {
```
Comment (oskardudycz) [inline src/__test__/projections/getProjectionStatus.test.ts:23]: **SUGGESTION:** I think it'd be more consistent if we use the same semantics in test names as in the method name.
```suggestion
describe("getProjectionStatus", () => {
```

File: src/__test__/projections/getProjectionStatus.test.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  describe("gets Projection Status", () => {
+    test("gets status", async () => {
+      for (const projection of projections) {
+        const status = await client.getProjectionStatus(projection);
+        expect(possibleStatuses).toContain(status);
+      }
+    });
+
+    test("non-existant", async () => {
```
Comment (oskardudycz) [inline src/__test__/projections/getProjectionStatus.test.ts:81]: **TYPO:**
```suggestion
    test("non-existent", async () => {
```

File: src/projections/getProjectionStatus.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  this: Client,
+  projectionName: string,
+  baseOptions: GetProjectionStatusOptions = {}
+): Promise<ProjectionStatus> {
+  debug.command("getProjectionStatus: %O", {
+    projectionName,
+    options: baseOptions,
+  });
+
+  const { projectionStatus } = await this.getProjectionStatistics(
```
Comment (oskardudycz) [inline src/projections/getProjectionStatus.ts:33]: **SUGGESTION:** Shouldn't we mark `getProjectionStatistics` and `GetProjectionStatisticsOptions` as obsolete/deprecated?

**Review by oskardudycz — CHANGES_REQUESTED** [review]
See the comments above.

File: src/__test__/projections/getProjectionStatus.test.ts (hunk 1 above)
Comment (George-Payne) [reply src/__test__/projections/getProjectionStatus.test.ts:23]: Copy paste issue :D

File: src/projections/getProjectionStatus.ts (hunk 3 above)
Comment (George-Payne) [reply src/projections/getProjectionStatus.ts:33]: They're still a supported operation: https://github.com/EventStore/architecture-and-planning/blob/main/rfcs/021%20-%20gRPC%20Projections.md#supported-operations

Comment (oskardudycz) [reply src/projections/getProjectionStatus.ts:33]: Hm 🤔 that's weird.

**Comment by George-Payne** [comment]
Some confusion around the RFC, will close this.

see: https://github.com/EventStore/architecture-and-planning/pull/100

**Pull request closed without merging**